qutebrowser/browser/urlmarks.py

```diff
@@ -218,6 +215,24 @@ def quickmark_del(self, name):
         except KeyError:
             raise cmdexc.CommandError("Quickmark '{}' not found!".format(name))
 
+    def get_by_qurl(self, url):
+        """Look up a quickmark by QUrl, returning its name.
+
+        Takes O(n) time, where n is the number of quickmarks.
+        Use a name instead where possible.
+        """
+        if not url.isValid():
+            raise ValueError("Invalid URL: {}".format(url.errorString()))
+        urlstr = url.toString(QUrl.RemovePassword | QUrl.FullyEncoded)
+
+        try:
+            index = list(self.marks.values()).index(urlstr)
+            key = list(self.marks.keys())[index]
+        except ValueError:
+            raise cmdexc.CommandError(
+                "Quickmark for '{}' not found!".format(urlstr))
```

Comment:
I try to avoid raising a `CommandError` from more generic methods (i.e. this could also be used outside a command). I think it'd be better to let the `KeyError` propagate and handle it in `quickmark_del` and `bookmark_del`.